File: Leetcode/PY/Leetcode825-适龄的朋友.py
class Solution:
    def numFriendRequests(self, ages: List[int]) -> int:
        n,cnt = len(ages),[0] * 121
        ans = n ** 2
        for age in ages: cnt[age] += 1
        for i in range(1,121):
            for j in range(1,121):
                if j <= 0.5 * i + 7 or j > i:
                    ans -= cnt[i] * cnt[j]
                elif i == j:
                    ans -= cnt[i]
        return ans

File: Leetcode/Rust/Leetcode825-适龄的朋友.rs
impl Solution {
    pub fn num_friend_requests(ages: Vec<i32>) -> i32 {
        let n = ages.len();
        let mut ans = (n * n) as i32;
        let mut cnt = vec![0;121];
        for age in ages.iter(){
            cnt[*age as usize] += 1;
        }
        for i in 1..=120{
            for j in 1..=120{
                if j > i || 2 * j <= i + 14 {
                    ans -= cnt[i] * cnt[j];
                }else if j == i{
                    ans -= cnt[i];
                }
            }
        }
        ans
    }
}